server/src-lib/Hasura/GraphQL/Transport/WebSocket/Protocol.hs

```diff
@@ -56,18 +56,19 @@ instance J.FromJSON ClientMsg where
   parseJSON = J.withObject "ClientMessage" $ \obj -> do
     t <- obj J..: "type"
     case t of
-      "connection_init" -> CMConnInit <$> obj J..:? "payload"
-      "start" -> CMStart <$> J.parseJSON (J.Object obj)
-      "stop" -> CMStop <$> J.parseJSON (J.Object obj)
+      "connection_init"      -> CMConnInit <$> obj J..:? "payload"
+      "start"                -> CMStart <$> J.parseJSON (J.Object obj)
+      "stop"                 -> CMStop <$> J.parseJSON (J.Object obj)
       "connection_terminate" -> return CMConnTerm
-      _ -> fail $ "unexpected type for ClientMessage: " <> t
+      _                      -> fail $ "unexpected type for ClientMessage: " <> t
 
 -- server to client messages
 
 data DataMsg
   = DataMsg
-  { _dmId      :: !OperationId
-  , _dmPayload :: !GraphqlResponse
+  { _dmId                 :: !OperationId
+  , _dmPayload            :: !GraphqlResponse
+  , _dmQueryExecutionTime :: !(Maybe Double)
   }
```

Comment:
This is the ADT which represents the data frame sent to the client as response. Are we sending the query execution time in the response as well?